components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/RequestFilterTest.java

```diff
@@ -62,16 +64,21 @@ public void testGetFilteredHeaders() {
         Mockito.when(config.getExcludedHeadersInActionRequestForActionType(ActionType.PRE_ISSUE_ACCESS_TOKEN))
                 .thenReturn(excludedHeaders);
 
-        Map<String, String[]> headers = new HashMap<>();
-        headers.put("Content-Type", new String[]{"application/json"});
-        headers.put("X-Header-1", new String[]{"X-header-1-value"});
-        headers.put("X-Header-3", new String[]{"X-header-3-value"});
+        List<Header> headers = new ArrayList<>();
+        headers.add(new Header("Content-Type", new String[]{"application/json"}));
+        headers.add(new Header("X-Header-1", new String[]{"X-header-1-value"}));
+        headers.add(new Header("X-Header-3", new String[]{"X-header-3-value"}));
 
-        Map<String, String[]> filteredHeaders =
+        List<Header> filteredHeaders =
                 RequestFilter.getFilteredHeaders(headers, ActionType.PRE_ISSUE_ACCESS_TOKEN);
         assertEquals(filteredHeaders.size(), 2);
-        assertEquals(filteredHeaders.get("Content-Type"), new String[]{"application/json"});
-        assertEquals(filteredHeaders.get("X-Header-3"), new String[]{"X-header-3-value"});
+        filteredHeaders.forEach(filteredHeader -> {
+            if (filteredHeader.getName().equalsIgnoreCase("Content-Type")) {
```

Comment:
Even here ignore case is not needed right ?
You just check if excluded headers are not available.

Reply:
This is updated